spec/amp-localstorage.md

```diff
@@ -0,0 +1,73 @@
+<!---
+Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# LocalStorage Usage in AMP
+
+AMP relies on the Viewer's support to use localStorage when the AMP page is embedded in a Viewer. This is because localStorage set within a cross domain iframe is not persisted in Safari.
+
+## AMP Viewer's Support
+
+The AMP Viewer declares its support to the localStorage API by setting the the url param `storage` to `true`. If so the Viewer must support the following APIs
+
+
+### Storage API
+
+Information entry can be passed between the AMP page and the supported AMP viewer through postMessages. And the AMP viewer will be responsible to store the info in the top level window localStorage.
+
+| Message             | Description | Request | Response                               |
+|---------------------| ------------|---------|----------------------------------------|
+|`loadStore`|Requests the local storage blob for the document's origin.|{'origin': (string)}|{'blob': (string)}|
+|`saveStore`|Requests that the specified blob is stored to local storage for the document's origin.|{'origin': (string), 'blob': (string)}|undefined (The response resolves once the blob has been successfully stored.)|
+
+### Data Format
+
+localStorage entry will be formatted as following
+```
+const entry = {
+  'key': {
+    'v': 'value',
+    't': timestamp,
+  }
+}
+blob = btoa(JSON.stringify(entry))
+```
+
+## Current Usage
+
+The following AMP components and service are using the localStorage.
+
+- `<amp-app-banner>` : Store the user decision on dismiss banner
+- `<amp-user-notification>` : Store the user decision on dismiss notification
+- `<amp-consent>` : Store the user decision and granular information on consent
```

Comment:
"any future usage need to be listed here"

Reply:
That's a good point. But really hard to keep the doc up to date though. 
Best I can do is to add one sentence to the last session.

Reply:
Please see if the new change looks good to you. thanks

Reply:
we can add a link in the presubmit warning:
https://github.com/ampproject/amphtml/blob/75e7d72be80d9bd356ad76e0a419cd2539320e6b/build-system/tasks/presubmit-checks.js#L27-L30

Reply:
👍 That's a great idea. Done, ptal